VS2022 bots are failing, spuriously?

**Description**
The VS2022 debug and releases CI bots are failing

**Steps to Reproduce**
* Any PR within the last two days fails.  I think this will happen with a no-change commit.

Example: the latest commit on main is https://github.com/microsoft/DirectXShaderCompiler/commit/8408ae8829c9c5c5e5e5c01ae8603d45538f9339 
It fails the Windows bots using VS2022
https://github.com/microsoft/DirectXShaderCompiler/runs/25879024728

The debug build fails with a compile error:

> 2024-06-06T06:24:48.6665848Z D:\a\1\s\projects\dxilconv\include\ShaderBinary\ShaderBinary.h(381,10): error C2220: the following warning is treated as an error [D:\a\1\projects\dxilconv\lib\ShaderBinary\ShaderBinary.vcxproj]
> 2024-06-06T06:24:48.7758196Z D:\a\1\s\projects\dxilconv\include\ShaderBinary\ShaderBinary.h(381,10): warning C4201: nonstandard extension used: nameless struct/union [D:\a\1\projects\dxilconv\lib\ShaderBinary\ShaderBinary.vcxproj]

That seems wrong because the code is explicitly guarded with a pragma to disable that warning.

The release build has a mysterious error message:

> C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'D:\a\1\CMakeFiles\c010341e1c4c2694685d486ffd93ffcb\StmtNodes.inc.tmp.rule;D:\a\1\CMakeFiles\c010341e1c4c2694685d486ffd93ffcb\StmtNodes.inc.rule;D:\a\1\CMakeFiles\e1327880013738d1b2f92ba750fb28de\ClangStmtNodes.rule;D:\a\1\s\tools\clang\include\clang\AST\CMakeLists.txt' exited with code -1073741819. [D:\a\1\tools\clang\include\clang\AST\ClangStmtNodes.vcxproj]

Is it being killed with the debug build fails?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,12 +120,6 @@ if(WIN32 AND NOT (CMAKE_C_COMPILER_ARCHITECTURE_ID MATCHES "ARM.*"))
   add_link_options(/CETCOMPAT)
 endif()
 
-# This is a hack workaround for the broken azure runner images:
-# https://github.com/actions/runner-images/issues/10004
-if(WIN32)
-  add_compile_definitions(_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR)
-endif()
-
 # HLSL Change Ends
 
 # HLSL Change Starts - set flag for Appveyor CI

